certdb/ocspstapling/ocspstapling_test.go

```diff
@@ -0,0 +1,173 @@
+)
+
+func TestStapleSCTList(t *testing.T) {
+	// Create a CA certificate
+	issuer, issuerPrivKey, err := makeCert(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a certificate for which to make an OCSP response
+	responderCert, _, err := makeCert(issuer)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an OCSP response
+	template := ocsp.Response{ // TODO: populate
+		SerialNumber: responderCert.SerialNumber,
+		IssuerHash:   crypto.SHA256,
+		Status:       ocsp.Good,
+	}
+
+	respDER, err := ocsp.CreateResponse(issuer, responderCert, template, issuerPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an empty DB of OCSP responses
+	// TODO: Test non-sqlite db's
+	gopath := os.Getenv("GOPATH")
+	dbPath := gopath + "/src/github.com/cloudflare/cfssl/certdb/testdb/certstore_development.db"
+	testDB := sql.NewAccessor(testdb.SQLiteDB(dbPath))
+
+	// Store the OCSP response in the DB
+	respSN := responderCert.SerialNumber.Text(16)
+	testDB.InsertOCSP(certdb.OCSPRecord{
+		Serial: respSN,
+		Body:   base64.StdEncoding.EncodeToString(respDER),
+		//Expiry:,
+	})
+
+	// Insert an SCT for the OCSP response we created
+	var zeroSCT ct.SignedCertificateTimestamp
+	err = StapleSCTList(testDB, respSN, "", []ct.SignedCertificateTimestamp{zeroSCT}, issuer, issuerPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the SCT was inserted
+	recs, err := testDB.GetOCSP(respSN, "")
+	if len(recs) == 0 {
+		t.Fatal("SCT could not be retrieved from DB:", zeroSCT)
+	}
+
+	respDER, err = base64.StdEncoding.DecodeString(recs[0].Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	response, err := ocsp.ParseResponse(respDER, issuer)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	scts := SCTListFromOCSPResponse(response)
+	if len(scts) == 0 {
+		t.Fatal("No SCTs in OCSP response:", response)
+	}
+
+	// TODO: Verify that scts[0] is equivalent to zeroSCT
+}
+
+// TODO: COMMENT
+var serialCounter int64
+
+// TODO: COMMENT
+func nextSN() *big.Int {
+	i := big.NewInt(serialCounter)
+	serialCounter++
+	return i
+}
+
+// TODO: COMMENT
+func makeCert(issuer *x509.Certificate) (*x509.Certificate, crypto.Signer, error) {
+	// Create a new private key
+	privKey, err := rsa.GenerateKey(rand.Reader, 1024)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	template := x509.Certificate{
+		SerialNumber: nextSN(),
+		Subject: pkix.Name{
+			Organization: []string{"Cornell CS 5152"},
+		},
+		AuthorityKeyId: []byte{42, 42, 42, 42},
+	}
+
+	if issuer == nil { // the cert is self-signed
+		issuer = &template
+	}
+
+	der, err := x509.CreateCertificate(rand.Reader, &template, issuer, privKey.Public(), privKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cert, privKey, nil
+}
+
+// SCTListFromOCSPResponse extracts the SCTList from an ocsp.Response
+// Returns an empty list if the SCT extension was not found or could not be
+// unmarshalled.
+// TODO: Move to helpers?
```

Comment:
You've got `SerializeSCTList` in helpers, so I think it makes sense to put this there, too.